pkg/v1/remote/descriptor.go

```diff
@@ -367,7 +368,7 @@ func (f *fetcher) fetchBlob(ctx context.Context, size int64, h v1.Hash) (io.Read
 
 	resp, err := f.Client.Do(req.WithContext(ctx))
 	if err != nil {
-		return nil, err
+		return nil, redact.Error(err)
```

Comment:
Maybe we want to do this everywhere we do `f.Client.Do` (or create a `f.Do` that does it)? Or at least in `headBlob` and `blobExists` as well, given we expect this to happen for redirected things, which are usually blobs.